src/main/java/com/puppycrawl/tools/checkstyle/checks/design/FinalClassCheck.java

```diff
@@ -220,6 +247,21 @@ private void registerNestedSubclassToOuterSuperClasses(DetailAST classAst) {
         }
     }
 
+    /**
+     * Check if class name matches with anonymous inner class name.
+     *
+     * @param ast current ast.
+     * @param classDesc class to match.
+     * @return true if current class name matches anonymous inner
+     *         class name.
+     */
+    private boolean doesNameOfClassMatchAnonymousInnerClassName(DetailAST ast,
+                                                               ClassDesc classDesc) {
+        final String classNameExtractor = Pattern.quote(packageName + PACKAGE_SEPARATOR) + "|\\.";
+        final String[] className = classDesc.getQualifiedName().split(classNameExtractor);
+        return ast.getFirstChild().getText().equals(className[className.length - 1]);
```

Comment:
Similar to https://github.com/checkstyle/checkstyle/pull/9383#discussion_r595431022 , why can't `classNameExtractor` just be `\\.` ? Can you show an example where we must look at package name?

Reply:
Yes, it can be, just wanted to keep the array size small ( which won't create a big difference, I know), changed it.